File: app/src/main/java/com/aquaguard/domain/repository/AlertRepository.java
package com.aquaguard.domain.repository;

import androidx.lifecycle.LiveData;
import com.aquaguard.domain.model.Alert;
import java.util.List;

public interface AlertRepository {
    LiveData<List<Alert>> getActiveAlerts(String deviceId);
    LiveData<List<Alert>> getAlertHistory(String deviceId);
    void createAlert(Alert alert, RepositoryCallback<Void> callback);
    void resolveAlert(String alertId, RepositoryCallback<Void> callback);
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/repository/AlertRepository.kt
package com.aquaguard.domain.repository

import com.aquaguard.domain.model.Alert
import kotlinx.coroutines.flow.Flow

interface AlertRepository {
    fun getActiveAlerts(deviceId: String): Flow<List<Alert>>
    fun getAlertHistory(deviceId: String): Flow<List<Alert>>
    suspend fun createAlert(alert: Alert): Result<Unit>
    suspend fun resolveAlert(alertId: String): Result<Unit>
}
